cli/internal/taskhash/taskhash_test.go

```diff
@@ -51,7 +51,7 @@ func Test_manuallyHashPackage(t *testing.T) {
 		t.Fatalf("failed to write contents to .gitignore: %v", err)
 	}
 	rootIgnoreFile.Close()
-	pkgIgnoreFilename := repoRoot.Join(pkgName, ".gitignore")
+	pkgIgnoreFilename := repoRoot.Join(turbopath.RelativeSystemPath(pkgName.ToString()), ".gitignore")
```

Comment:
(request for change)

This isn't supposed to work, I'm not actually sure why it does? Is that literal `.gitignore` somehow getting past the type check?

```go
func (p AbsoluteSystemPath) Join(additional ...RelativeSystemPath) AbsoluteSystemPath {
	cast := RelativeSystemPathArray(additional)
	return AbsoluteSystemPath(filepath.Join(p.ToString(), filepath.Join(cast.ToStringArray()...)))
}

pkgIgnoreFilename := repoRoot.Join( // AbsoluteSystemPath
	turbopath.RelativeSystemPath(
		pkgName.ToString() // AnchoredSystemPath
	),
	".gitignore" // string, wat?
)
```

The way I envisioned doing this:
```go
pkgName.RestoreAnchor(repoRoot).Join(turbopath.RelativeSystemPath(".gitignore");
```

Reply:
literals are a hole in Go's newtype implementation. They are implicitly treated as the expected type.